kotest-assertions/kotest-assertions-json/src/jvmTest/kotlin/com/sksamuel/kotest/tests/json/ContainJsonKeyValueTest.kt

```diff
@@ -55,11 +56,19 @@ class ContainJsonKeyValueTest : StringSpec({
       """.trimIndent()
    }
 
+   "Failure message states if key is missing, shows valid subpath with lower array index" {
+      shouldFail {
+         json.shouldContainJsonKeyValue("$.store.book[1].tags[2]", "humor")
+      }.message shouldBe """
+         Expected given to contain json key <'$.store.book[1].tags[2]'> but key was not found. Found valid path: <'$.store.book[1].tags[1]'>
```

Comment:
I'm wondering if we should take an alternate approach here?

If we're looking for an array element, and we fail to find that subpath, couldn't we just try to get an `Array<Any> using JsonPath for that path? If successful, we print: 
```
Expected array at path <'$currentSubPath'> to have $expectedElementCount, but actual count is 
$actualElementCount
```

If we fail to get an array at all, we could immediately remove last part of the path and move on.

Reply:
> I'm wondering if we should take an alternate approach here?
> 
> If we're looking for an array element, and we fail to find that subpath, couldn't we just try to get an `Array using JsonPath for that path? If successful, we print:
> 
> ```
> Expected array at path <'$currentSubPath'> to have $expectedElementCount, but actual count is 
> $actualElementCount
> ```
> 
> If we fail to get an array at all, we could immediately remove last part of the path and move on.

makes sense. will do